src/components/ItemFilter.vue

```diff
@@ -121,33 +168,53 @@ export default {
             return dt;
           });
         }
-        if (this.sort && typeof filters.sortby === 'string') {
-          this.sortOrder = filters.sortby.startsWith('-') ? -1 : 1;
-          this.sortTerm = filters.sortby.replace(/^(\+|-)/, '');
-        }
         this.filters = filters;
       }
     }
+  },
+   mounted () {
+    this.createBlankSearchFilter();
   },
   methods: {
+    sortFieldSet (value) {
+      this.sortTerm = value;
+      this.queryableSet(this.itemSearch.sortFragment.queryable, value);
+    },
+    sortDirectionSet (value) {
+      this.sortOrder = value;
+      this.itemSearch.sortFragment.direction = value;
+    },
+    queryableSet (item, event) {
+      item.props.value = event;
+    },
+    removeUserFilterField (item) {
+      this.itemSearch.filterFragment.removeQueryableInput(item);
+    },
+    additionalFieldSelected (selected) {
+      const queryable = this.itemSearch.filterFragment.queryables.find(q => q.id === selected.value);
+      this.itemSearch.filterFragment.createQueryableInput(queryable);
+    },
     getDefaultValues() {
       return {
         datetime: null,
         bbox: null,
         limit: null,
         ids: [],
         collections: [],
-        sortby: null
+        sortby: null,
+        advancedFilters: null
       };
     },
     onSubmit() {
       if (this.sort) {
         this.filters.sortby = this.formatSort();
       }
+      this.filters.advancedFilters = this.itemSearch.getAsCql2Json();
```

Comment:
In global search I'm getting for this line:

> TypeError: this.itemSearch is null

Reply:
See comment above, moving onto global search now.